app/templates/courses/teacher-class-view.jade

```diff
@@ -346,6 +346,8 @@ mixin copyCodes
   div.copy-button-group.form-inline.m-b-3
     .form-group
       input.text-h4.semibold#join-code-input(value=state.get('classCode'), readonly)
+    button#regenerate-code-btn.form-control.btn.btn-lg.btn-forest(disabled=!Boolean(value=state.get('classCode')))
+      span(data-i18n='teacher.regenerate_class_code') Regenerate
```

Comment:
We will need to add new strings to the `en.coffee` locale file in the right place. We also now prefer to leave the literal text out of the template, since sometimes it confuses us whether we are going to show that text or whatever's in the i18n (which might have changed, or which we might have forgotten to add.)

```suggestion
      span(data-i18n='teacher.regenerate_class_code')
```